packages/python-google-compute-engine/google_compute_engine/instance_setup/instance_setup.py

```diff
@@ -65,17 +66,27 @@ def __init__(self, debug=False):
       instance_config_metadata = self._GetInstanceConfig()
       self.instance_config = instance_config.InstanceConfig(
           logger=self.logger, instance_config_metadata=instance_config_metadata)
+
       if self.instance_config.GetOptionBool('InstanceSetup', 'set_host_keys'):
         host_key_types = self.instance_config.GetOptionString(
             'InstanceSetup', 'host_key_types')
         self._SetSshHostKeys(host_key_types=host_key_types)
+
       if self.instance_config.GetOptionBool('InstanceSetup', 'set_boto_config'):
         self._SetupBotoConfig()
+
+      # machineType is e.g. 'projects/00000000000000/machineTypes/n1-standard-1'
+      machineType = self.metadata_dict['instance']['machineType'].split('/')[-1]
```

Comment:
I think this belongs in a separate function. I don't see where we are currently testing that the machine type will be read correctly from metadata. Also, as written, I believe the current tests will be calling `subprocess.call(['sysctl', 'vm.overcommit_memory=1'])` since subprocess is not stubbed in many of the instance setup tests calling this.

Reply:
Actually, because the fake metadata in the tests either doesn't have machine type, or has machine type with a value that does not start with `e2-` they won't be calling subprocess. Still doesn't seem ideal to me.

Reply:
Something like this:
```
def _DisableOvercommit(self):
  """Disable overcommit accounting on E2 machine types."""

  # Expected machine type format:
  # 'projects/00000000000/machineTypes/n1-standard-1'.
  machine_type = self.metadata_dict['instance']['machineType'].split('/')[-1]
  if machine_type.startswith('e2-') and 'bsd' not in distro_name:
    subprocess.call(['sysctl', 'vm.overcommit_memory=1'])
```